builtin/game/item_entity.lua

```diff
@@ -148,49 +148,46 @@ core.register_entity(":__builtin:item", {
 		local vel = self.object:getvelocity()
 		local def = node and core.registered_nodes[node.name]
 		-- Ignore is nil -> stop until the block loaded
-		local is_physical = (def and not def.walkable) or vel.y ~= 0
+		local is_moving = (def and not def.walkable) or
+				(vel.x ~= 0 or vel.y ~= 0 or vel.z ~= 0)
```

Comment:
Brackets not needed, might be faster without them

Reply:
Should I remove the brackets around "def and not def.walkable" as well?

Reply:
No, because there they are needed to define the order of computation: AND NOT first, then OR OR OR.

Reply:
Ironically you were totally right about the first comment to remove \*brackets\* as the logical AND has a higher priority than OR. Definitions. JFYI.
```Lua
> return ({} and not true) or (0 ~= 0 or -5 ~= 0 or 0 ~= 0)
true
> return {} and not true or 0 ~= 0 or -5 ~= 0 or 0 ~= 0
true
> return ({} and not true) or (0 ~= 0 or 0 ~= 0 or 0 ~= 0)
false
> return {} and not true or 0 ~= 0 or 0 ~= 0 or 0 ~= 0
false
```

Reply:
The precedence of not is higher than the one of and, which is higher than the one of or, right?
But that would be confusing, I'll leave them.

Reply:
Yeah, but AND having higher precedence than OR is not obvious, i forget that all the time, so yes it's there to make precedence obvious and clear, whereas the brackets around the 'vel's do not help with clarity at all and are actually confusing.

Reply:
http://www.lua.org/manual/5.1/manual.html#2.5.6